data/readreplica.py

```diff
@@ -71,6 +71,33 @@ def execute_sql(self, sql, params=None, commit=SENTINEL):
                     raise
 
 
+class DoubleWriteWrapper(AutomaticFailoverWrapper):
+
+    def __init__(self, primary_db, fallback_db, secondary_write_db=None):
+        super(AutomaticFailoverWrapper, self).__init__(primary_db, fallback_db)
+        self._secondary_write_db = secondary_write_db
+
+    def __getattr__(self, attribute):
+        return super(AutomaticFailoverWrapper, self).__getattr__(attribute)
+
+    def execute(self, query, commit=SENTINEL, **context_options):
+        ctx = self.get_sql_context(**context_options)
+        sql, params = ctx.sql(query).query()
+        return self.execute_sql(sql, params, commit=commit)
+
+    def execute_sql(self, sql, params=None, commit=SENTINEL):
+        result = super(AutomaticFailoverWrapper, self).execute_sql(sql, params, commit)
```

Comment:
Does `execute_sql` execute a full transaction or can we have cases like `START TRANSACTION` and `COMMIT` are spred over multiple `execute_sql` calls?

Reply:
I believe the transaction is outside of this for the time being. That would be something to add to allow a rollback of the primary db when we are ready to tie both database commits together.

Reply:
How do you make sure that the primary key IDs are same across both the database. If the sql adds a new row in DB1 how are you making sure the same ID is going to be used for DB2

Reply:
Currently we have no way to guarantee the values stay the same, thus the argument is weak, however the nature of the schema will want to stay in sync without failure writes, which in phase 2 we will guarantee if 1 write fails they both do. Secondly we only join on ID and find it with references through other means which means that in theory even if the IDs become different the reads will still succeed as they did before.